### PR TITLE
Fat search api / google vision / video upload / image upload

### PR DESCRIPTION
**Pull request by calmoon808**
You can test the components in playground. Google vision only works in conjunction with fat search api, but fat search api can work without google vision. 

Video/Image upload components only puts the video/image url into store right now.

Also, if you're trying to test the new api I'm gonna need the IP address of the current network you're on that appears in the console.log from the actionsFatSecretFoodSearch response.

### CODE REVIEW

**Comment by calmoon808** [comment]
Also, if you're trying to test the new api I'm gonna need the IP address of the current network you're on that appears in the console.log from the actionsFatSecretFoodSearch response.

File: public/index.html [hunk 1]
```diff
@@ -41,4 +41,10 @@
       To create a production bundle, use `npm run build` or `yarn build`.
     -->
   </body>
+  <!-- <script src="https://platform.fatsecret.com/js?key=2f6a2dd0d2d94154a9cfc666ac471148"></script>
+  <script>
+    console.log(fatsecret);
+    fatsecret.setContainer("fatsecret_container");
+    fatsecret.setCanvas("home");
+  </script> -->
```
Comment (danieltichiyama) [inline public/index.html]: necessary?

File: server/api/exercise/tutorial_videos/index.js [hunk 2]
```diff
@@ -1,12 +1,28 @@
...
+        return res.json(response);
+      });
+    }
+  )
+  .post((req, res) => {
+    console.log(req.body);
+    return req.db.TutorialVideo.forge(req.body)
+    .save()
+    .then(results => {
+      console.log('VIDEO POST', results);
```
Comment (danieltichiyama) [inline server/api/exercise/tutorial_videos/index.js]: necessary?

File: server/api/exercise/tutorial_videos/index.js [hunk 3]
```diff
@@ -1,12 +1,28 @@
...
+  .get((req, res) => {
+    return req.db.TutorialVideo.fetchAll({
+        withRelated: ["exercise_id"]
+      }).then(response => {
+        return res.json(response);
+      });
+    }
+  )
+  .post((req, res) => {
+    console.log(req.body);
```
Comment (danieltichiyama) [inline server/api/exercise/tutorial_videos/index.js]: necessary?

File: server/api/nutrition/fat_secret/index.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
+const express = require("express");
+const fatSecretRouter = express.Router();
+const request = require("request");
+const axios = require("axios");
+const clientID = process.env.REACT_APP_FAT_SECRET_CLIENT_ID;
+const clientPW = process.env.REACT_APP_FAT_SECRET_CLIENT_SECRET;
```
Comment (danieltichiyama) [inline server/api/nutrition/fat_secret/index.js:6]: could you please give these to us, so we can add it to our ENV files?

File: server/api/nutrition/fat_secret/index.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+          })
+          .save()
+          .then(() => {
+            //set timer to refresh token;
+            timer = setInterval(() => {
+              fatSecretApi();
+            }, 82,800,000)
+          });
+        };
+      });
```
Comment (danieltichiyama) [inline server/api/nutrition/fat_secret/index.js:64]: bomb ass code broh.

File: server/api/nutrition/fat_secret/index.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+      //stop timer then re run function
+      .then(() => {
+        clearInterval(timer);
+        fatSecretApi();
+      })
+    }
+  })
+}
+
+fatSecretApi();
```
Comment (danieltichiyama) [inline server/api/nutrition/fat_secret/index.js:81]: just wanna make sure, this shouldn't run every time we make a call to this route, right?  I will check when I try it out, but if you could confirm before I have a chance.

File: server/api/nutrition/nutrition/index.js [hunk 7]
```diff
@@ -36,33 +37,39 @@ nutritionRouter
...
-      return res.json(response.data);
-    })
-    .catch(err => {
-      console.log(err);
-    });
+  .then(response => {
+    return res.json(response.data);
+  })
+  .catch(err => {
+    console.log('calvin is doodooboy', err);
```
Comment (danieltichiyama) [inline server/api/nutrition/nutrition/index.js:71]: hahahaha

File: server/api/nutrition/vision/index.js [hunk 8]
```diff
@@ -5,19 +5,26 @@ const path = require('path');
...
     // Creates a client
     const client = new vision.ImageAnnotatorClient({ keyFilename });
     // Performs label detection on the image file
     const [result] = await client.labelDetection(req.file.buffer);
     const labels = result.labelAnnotations;
-    console.log('Labels:');
-    labels.forEach(label => console.log(label.description));
+    const filterLabel = labels.filter(label => !filterArr.includes(label.description));
+    // console.log('Labels:');
+    // filterLabel.forEach(label => console.log(label.description));
```
Comment (danieltichiyama) [inline server/api/nutrition/vision/index.js:20]: remove if unnecessary.

File: server/api/upload/image_upload/index.js [hunk 9]
```diff
@@ -0,0 +1,67 @@
+const express = require("express");
+const imageUploadRouter = express.Router();
+const multer = require("multer");
+const multerS3 = require("multer-s3");
+const path = require('path');
+const aws = require("aws-sdk");
+
+const s3 = new aws.S3({
+  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+  accessKeyId: process.env.AWS_ACCESS_KEY,
```
Comment (danieltichiyama) [inline server/api/upload/image_upload/index.js:10]: we need this in our ENV for it to work?

File: server/api/upload/video_upload/index.js [hunk 10]
```diff
@@ -0,0 +1,67 @@
+const express = require("express");
+const videoUploadRouter = express.Router();
+const multer = require("multer");
+const multerS3 = require("multer-s3");
+const path = require('path');
+const aws = require("aws-sdk");
+
+const s3 = new aws.S3({
+  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+  accessKeyId: process.env.AWS_ACCESS_KEY,
```
Comment (danieltichiyama) [inline server/api/upload/video_upload/index.js:10]: need this in our ENV?

File: server/database/decorator.js [hunk 11]
```diff
@@ -18,6 +18,8 @@ const User = require("./models/User");
 const UserTier = require("./models/UserTier");
 const UserVideo = require("./models/UserVideo");
 const Workout = require("./models/Workout");
+// const WorkoutExercise = require("./models/WorkoutExercise");
```
Comment (danieltichiyama) [inline server/database/decorator.js:21]: rm if unnecessary

File: src/actions/index.js [hunk 12]
```diff
@@ -126,8 +130,10 @@ export const actionsLoadActivity = () => async dispatch => {
 };
 
 export const actionsLoginSubmit = data => async dispatch => {
+  console.log(data);
```
Comment (danieltichiyama) [inline src/actions/index.js:133]: rm

File: src/actions/index.js [hunk 13]
```diff
@@ -126,8 +130,10 @@ export const actionsLoadActivity = () => async dispatch => {
...
   await Axios.post("/api/auth/login", data)
     .then(response => {
+      console.log(response);
```
Comment (danieltichiyama) [inline src/actions/index.js:136]: rm

File: src/actions/index.js [hunk 14]
```diff
@@ -338,6 +345,29 @@ export const actionsLoadExerciseList = () => async dispatch => {
...
+  await Axios.post("/api/fat_secret/nutrients", {data: data})
+    .then(response => {
+      return dispatch({
+        type: FAT_SECRET_FOOD_NUTRIENT_SEARCH,
+        payload: response.data.food
+      })
+    })
+}
 export const actionsDeleteFood = data => async dispatch => {
   console.log("data: ", data);
```
Comment (danieltichiyama) [inline src/actions/index.js:372]: rm

File: src/actions/index.js [hunk 15]
```diff
@@ -351,3 +381,29 @@ export const actionsDeleteFood = data => async dispatch => {
       console.log("Error in actionsDeleteFood: ", err);
     });
 };
+export const actionVideoUpload = data => async dispatch => {
+  await Axios.post("/api/video_upload", data)
+  .then(response => {
+    console.log(response);
```
Comment (danieltichiyama) [inline src/actions/index.js:387]: rm?

File: src/components/FoodSearchComponent/FoodSearchComponent.jsx [hunk 16]
```diff
@@ -9,22 +9,21 @@ class FoodSearchComponent extends Component {
...
 
-  async searchKeyword(e) {
+  searchKeyword = async (e) => {
     await this.setState({ data: e.target.value });
     // UNCOMMENT FOR LIVE SEARCHBAR
     // this.props.dispatchFoodSearch(this.state);
   }
 
   mapFoods = data => {
+    console.log('DATA: ', data);
```
Comment (danieltichiyama) [inline src/components/FoodSearchComponent/FoodSearchComponent.jsx:26]: rm?

File: src/components/FoodVisionComponent/FoodVisionComponent.jsx [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+  }
+
+  render() { 
+    return ( 
+      <div>
+          <input
+            type="file"
+            name="foodImage"
+            accept="image/*"
+            // className={styles.upload}
```
Comment (danieltichiyama) [inline src/components/FoodVisionComponent/FoodVisionComponent.jsx]: rm?

File: src/reducers/index.js [hunk 18]
```diff
@@ -84,11 +94,19 @@ let reducer = (store = initialStore, action) => {
       return Object.assign({}, store, { community_posts: action.payload });
 
     case FOOD_SEARCH:
+      console.log(action.payload.foods);
```
Comment (danieltichiyama) [inline src/reducers/index.js]: rm?

File: src/reducers/index.js [hunk 19]
```diff
@@ -134,6 +165,14 @@ let reducer = (store = initialStore, action) => {
     case LOAD_WORKOUTS:
       return Object.assign({}, store, { workouts: action.payload });
 
+    case VIDEO_UPLOAD:
+      console.log(action.payload);
```
Comment (danieltichiyama) [inline src/reducers/index.js]: rm?

File: src/reducers/index.js [hunk 20]
```diff
@@ -134,6 +165,14 @@ let reducer = (store = initialStore, action) => {
...
+      return Object.assign({}, store, { videos: action.payload });
+
+    case IMAGE_UPLOAD:
+      console.log(action.payload);
```
Comment (danieltichiyama) [inline src/reducers/index.js]: rm?

File: src/components/LabelComponent/LabelComponent.jsx [hunk 21]
```diff
@@ -0,0 +1,34 @@
+import React, { Component } from "react";
```
Comment (danieltichiyama) [inline src/components/LabelComponent/LabelComponent.jsx:1]: sorry, not sure where this component gets used...?

**Review by danieltichiyama — CHANGES_REQUESTED** [review]

File: public/index.html (hunk 1 above)
Comment (calmoon808) [reply public/index.html]: nope

File: server/api/exercise/tutorial_videos/index.js (hunk 3 above)
Comment (calmoon808) [reply server/api/exercise/tutorial_videos/index.js]: nope

File: server/api/nutrition/fat_secret/index.js (hunk 4 above)
Comment (calmoon808) [reply server/api/nutrition/fat_secret/index.js:6]: just posted to slack

File: server/api/nutrition/fat_secret/index.js (hunk 5 above)
Comment (calmoon808) [reply server/api/nutrition/fat_secret/index.js:64]: ty

File: src/components/FoodVisionComponent/FoodVisionComponent.jsx (hunk 17 above)
Comment (calmoon808) [reply src/components/FoodVisionComponent/FoodVisionComponent.jsx]: yep

File: src/reducers/index.js (hunk 18 above)
Comment (calmoon808) [reply src/reducers/index.js]: yep

File: src/reducers/index.js (hunk 19 above)
Comment (calmoon808) [reply src/reducers/index.js]: yep

File: src/reducers/index.js (hunk 20 above)
Comment (calmoon808) [reply src/reducers/index.js]: yep

File: src/components/LabelComponent/LabelComponent.jsx (hunk 21 above)
Comment (calmoon808) [reply src/components/LabelComponent/LabelComponent.jsx:1]: I'm still not 100% sure on what extending component does. I've been just putting that on all my components lol

File: server/api/exercise/tutorial_videos/index.js (hunk 2 above)
Comment (calmoon808) [reply server/api/exercise/tutorial_videos/index.js]: nope

**Comment by calmoon808** [comment]
changes are made

**Review by danieltichiyama — COMMENTED** [review]
Okay, image upload works, just waiting on the API IP address approval to test the search function.  I'm thinking about how the image upload should function.  Is it possible to store the image until after they choose what food it is, and upon choosing to add the food to a meal, could we then prompt them to save the image then?

**Review by danieltichiyama — APPROVED** [review]

**Pull request merged**